libs/ui-lib/lib/ocm/components/clusterWizard/OperatorsSelect.tsx

```diff
@@ -70,40 +75,93 @@ const OperatorsSelect = ({
       });
   }, [isSingleClusterFeatureEnabled, supportedOperators, opSpecs]);
 
-  if (isLoading) {
-    return <LoadingState />;
-  }
+  const bundledOperatorIds = bundles.flatMap((bundle) => bundle.operators || []);
+
+  const filteredOperators = React.useMemo(() => {
+    const inBundles = new Set(bundledOperatorIds);
+
+    return operators.filter((op) => {
+      const spec = opSpecs[op];
+      if (!spec) return false;
+
+      const title = spec.title?.toLowerCase() || '';
+      const description = spec.descriptionText?.toLowerCase() || '';
+      const matchesSearch =
+        searchTerm.trim() === '' ||
+        op.toLowerCase().includes(searchTerm.toLowerCase()) ||
+        title.includes(searchTerm.toLowerCase()) ||
+        description.includes(searchTerm.toLowerCase());
+
+      const isInBundle = inBundles.has(op);
+
+      return (
+        (isInBundle || matchesSearch) &&
+        (!isSingleClusterFeatureEnabled || singleClusterOperators.includes(op))
+      );
+    });
+  }, [bundledOperatorIds, operators, opSpecs, searchTerm, isSingleClusterFeatureEnabled]);
 
   const selectedOperators = values.selectedOperators.filter(
```

Comment:
The selection is not right, I believe is due to the fact that you're using the `filteredBundles` and `filteredOperators`
1. I select the OpenShift AI bundle, which includes the NVIDIA operator
2. I type "Nvi", the operator is shown but as if it was deselected.

![operators-selected](https://github.com/user-attachments/assets/af13c007-a482-4aa5-aa71-ddea17ef03bf)
![operators-not-selected](https://github.com/user-attachments/assets/db541d11-709c-4727-84f1-5abb8fbb7521)

Also note that the number of selected operators shown when filters are applied is incorrect.
IMO we should always keep showing the number of selected operators irrespective of the filter we're showing.